hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/functional/FutureIO.java

```diff
@@ -141,9 +145,11 @@ public static <T> List<T> awaitFuture(final Collection<Future<T>> collection)
       }
       return results;
     } catch (InterruptedException e) {
+      LOG.error("Execution of future interrupted ", e);
       throw (InterruptedIOException) new InterruptedIOException(e.toString())
           .initCause(e);
     } catch (ExecutionException e) {
+      LOG.error("Execution of future failed with exception", e.getCause());
```

Comment:
log this at debug. handlers up the stack can choose what to do -it may be harmless